ImportError: cannot import name 'FormattedExcinfo' on pytest 9.1.0

### Description

Tavern fails to import (and therefore breaks pytest collection entirely) when used with **pytest 9.1.0**. pytest 9.1 removed `FormattedExcinfo` from `_pytest._code.code`, but Tavern still imports it unconditionally:

```python
# tavern/_core/pytest/error.py
from _pytest._code.code import FormattedExcinfo, TerminalRepr
```

Since Tavern registers a pytest hook (`pytest_collect_file`), this import runs during collection and aborts the whole test session, not just Tavern tests.

### Traceback

```
.../tavern/_core/pytest/error.py:11: in <module>
    from _pytest._code.code import FormattedExcinfo, TerminalRepr
E   ImportError: cannot import name 'FormattedExcinfo' from '_pytest._code.code'
```

### Environment

- Tavern: 3.5.0 (also present in 3.6.0 — the import line is unchanged)
- pytest: 9.1.0 (works on 9.0.x, where `FormattedExcinfo` still exists)
- Python: 3.11 / 3.12 / 3.13 / 3.14

### Steps to reproduce

1. `pip install tavern pytest==9.1.0`
2. Run `pytest` in any project that collects Tavern YAML files.

### Impact

`pytest<10,>=8` is declared as compatible in Tavern's metadata, but 9.1.0 is in that range and is broken. Downstream projects are currently forced to pin `pytest<9.1` as a workaround.

### Suggested fix

`FormattedExcinfo` is a private (`_pytest`) API. Either guard the import for pytest >= 9.1, or avoid depending on it. If only `TerminalRepr` is needed at module scope, the `FormattedExcinfo` usage could be made lazy / version-conditional.

--- a/tavern/_core/pytest/error.py
+++ b/tavern/_core/pytest/error.py
@@ -13,7 +13,7 @@ from _pytest._code.code import TerminalRepr
 try:
     from _pytest._code.code import FormattedExcinfo
 except ImportError:
-    from pytest import ExceptionInfo as FormattedExcinfo
+    from _pytest._code.code import ExceptionInfoFormatter as FormattedExcinfo
 from _pytest._io import TerminalWriter
 
 from tavern._core import exceptions

--- a/tavern/_core/pytest/error.py
+++ b/tavern/_core/pytest/error.py
@@ -8,7 +8,12 @@ from io import StringIO
 from typing import Any
 
 import yaml
-from _pytest._code.code import FormattedExcinfo, TerminalRepr
+from _pytest._code.code import TerminalRepr
+
+try:
+    from _pytest._code.code import FormattedExcinfo
+except ImportError:
+    from pytest import ExceptionInfo as FormattedExcinfo
 from _pytest._io import TerminalWriter
 
 from tavern._core import exceptions

